Kudu.Core/Deployment/StructuredTextLogger.cs

```diff
@@ -0,0 +1,131 @@
+﻿using Kudu.Core.Tracing;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Kudu.Core.Deployment
+{
+    public class StructuredTextLogger : IDetailedLogger
+    {
+        private readonly static object DocumentLock = new object();
```

Comment:
i see. and we only have one deployment happen at a time, so the lock object won`t be share amount multiple files.

still, it won`t be ideal if we use it in scenario where we want to write message into multiple files at the same time. 

and this lock object seems specific to deployment log scenario. i would suggest either we rename the file to be more deployment specific or have a way to do lock per file for general case (maybe use a static Dictionary<file path, lock object> to store lock objects, but need to make sure we have another routine to clean invalid data from the Dictionary where log file haven`t been touched anymore or deleted).  

what do you think?